src/cluster/replication.cc

```diff
@@ -770,14 +770,16 @@ Status ReplicationThread::parallelFetchFile(const std::string &dir,
           if (this->stop_flag_) {
             return {Status::NotOK, "replication thread was stopped"};
           }
+          int sock_timeout_ms = 3000;
           ssl_st *ssl = nullptr;
 #ifdef ENABLE_OPENSSL
           if (this->srv_->GetConfig()->tls_replication) {
             ssl = SSL_new(this->srv_->ssl_ctx.get());
           }
           auto exit = MakeScopeExit([ssl] { SSL_free(ssl); });
 #endif
-          int sock_fd = GET_OR_RET(util::SockConnect(this->host_, this->port_, ssl).Prefixed("connect the server err"));
+          int sock_fd = GET_OR_RET(util::SockConnect(this->host_, this->port_, ssl, sock_timeout_ms, sock_timeout_ms)
```

Comment:
- Why have the connection timeout and the timeout both been set to the same value?
- Also why is the timeout set to 3000 here?
- Should we make this configurable?

Reply:
> I suggest separating those two default values due to their backoff interval is different. For the TCP `syn` packet loss retry in Linux is `1s`, `2s`, `4s`, `8s`, so a value like `3200ms` will be better than `3000ms` because it would retry twice if it's loss due to high traffic load.
> 
> And for the read/write backoff is related to RTO, but it's generally around 200ms in the internal network, so its backoff would be `200ms`, `400ms`, `800ms`, `1600ms`, so it will be better to set it to a value like `3100ms`.

updated

Reply:
> * Why have the connection timeout and the timeout both been set to the same value?
> * Also why is the timeout set to 3000 here?
> * Should we make this configurable?

I will make timeout config fields configurable, and using the default value suggested by hulk :)